### PR TITLE
specify type of dim and sym arrays

### LINKED ISSUE #88: MethodError for empty edge list
In the last CI run for PowerDynamics we got an error in two specific tests. Both tests are designed in a way, that the edge list is actually empty. We can easily circumvent this problem by adding lines to the testcase and everything runs smooth. But I assume that something changed in the last release of NetworkDynamics, such that you cannot hand over an empty edge list anymore. Here is a minimal example:

```julia
using NetworkDynamics, OrdinaryDiffEq, LightGraphs

N = 4
g = SimpleGraph(N)

function kuramoto_vertex!(dθ, θ, edges, ω, t)
    dθ[1] = ω
    sum_coupling!(dθ, edges)
end

edge_array = []

vertex! = ODEVertex(f! = kuramoto_vertex!, dim = 1, sym=[:θ])
nd! = network_dynamics(vertex!, edge_array, g)
```
This give the following error when executing the `network_dynamics()` function:
```julia
ERROR: LoadError: MethodError: no method matching zero(::Type{Any})
```

I'm not sure if this is case we actually want to cover, but apparently it worked before.

### PR DESCRIPTION
**Pull request by lindnemi**
fix #88

### CODE REVIEW

File: src/NetworkDynamics.jl [hunk 1]
```diff
@@ -36,27 +36,35 @@ constructors.
 function collect_ve_info(vertices!, edges!, graph)
     if vertices! isa Array
         @assert length(vertices!) == nv(graph)
-        v_dims = [v.dim for v in vertices!]
-        symbols_v = [Symbol(vertices![i].sym[j],"_",i) for i in 1:length(vertices!) for j in 1:v_dims[i]]
+        v_dims = Int64[v.dim for v in vertices!]
```
Comment (hexaeder) [inline src/NetworkDynamics.jl:39]:
```suggestion
        v_dims = Int[v.dim for v in vertices!]
```

File: src/NetworkDynamics.jl [hunk 2]
```diff
@@ -36,27 +36,35 @@ constructors.
...
+        symbols_v = Symbol[Symbol(vertices![i].sym[j],"_",i)
```
Comment (hexaeder) [inline src/NetworkDynamics.jl:40]:
```suggestion
        symbols_v = [Symbol(vertices![i].sym[j],"_",i)
```
This is a bit redundant. The `Symbol` constructor can be inferred to create symbols (even for empty edges `symbols_e isa Symbol[]` before this pr. But doesn't hurt either 🤷‍♂️

File: src/NetworkDynamics.jl [hunk 3]
```diff
@@ -36,27 +36,35 @@ constructors.
...
+                     for i in 1:nv(graph)
+                     for j in 1:v_dims[i]]
         mmv_array = [vertices!.mass_matrix for v in vertices(graph)]
     end
 
     if edges! isa Array
         @assert length(edges!) == ne(graph)
-        e_dims = [e.dim for e in edges!]
-        symbols_e = [Symbol(edges![i].sym[j],"_",i) for i in 1:length(edges!) for j in 1:e_dims[i]]
+        e_dims = Int64[e.dim for e in edges!]
```
Comment (hexaeder) [inline src/NetworkDynamics.jl:54]:
```suggestion
        e_dims = Int[e.dim for e in edges!]
```

File: src/NetworkDynamics.jl [hunk 4]
```diff
@@ -36,27 +36,35 @@ constructors.
...
+        symbols_e = Symbol[Symbol(edges![i].sym[j],"_",i)
```
Comment (hexaeder) [inline src/NetworkDynamics.jl:55]:
```suggestion
        symbols_e = [Symbol(edges![i].sym[j],"_",i)
```

**Review by antonplietzsch — APPROVED** [review]
The PD tests pass using this ND branch.

**Comment by lindnemi** [comment]
Thanks for the corrections!

**Pull request merged**